src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java

```diff
@@ -139,10 +139,10 @@ private BitbucketWebhook process(WebhookRegisterRequest request,
                                      Collection<BitbucketWebhookEvent> events) {
         String callback = constructCallbackUrl(request);
         List<BitbucketWebhook> webhooks = webhookClient.getWebhooks().collect(Collectors.toList());
-        Set<BitbucketWebhook> serverSideWebhooks =
+        List<BitbucketWebhook> serverSideWebhooks =
                 webhooks.stream()
                         .filter(hook -> hook.getName().equals(request.getName()) && hook.getUrl().equals(callback))
-                        .collect(Collectors.toSet());
+                        .collect(Collectors.toList());
```

Comment:
This is built around the assumption that the `webhooks` will not contain duplicate webhooks. I think that is a safe assumption.

Reply:
You could alternatively set the parameter type on `update` to a Collection, and do a bit of stream wrangling to pull out the first element, but I think this is fine.

Reply:
_Probably_ I need it as a list for the update later on, so I figured there was no reason to not use a list.